vm/actor/src/builtin/verifreg/mod.rs

```diff
@@ -0,0 +1,302 @@
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*STORAGE_MARKET_ACTOR_ADDR))?;
+        if params.deal_size < MINIMUM_VERIFIED_SIZE.into() {
+            return Err(ActorError::new(
+                ExitCode::ErrIllegalState,
+                format!(
+                    "Verified Dealsize {:} is below minimum in usedbytes",
+                    params.deal_size
+                ),
+            ));
+        }
+
+        rt.transaction::<_, Result<_, ActorError>, _>(|st: &mut State, rt| {
+            let verifier_cap = st
+                .get_verifier(rt.store(), &params.address)
+                .map_err(|_| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("Failed to get Verifier {:?}", &params.address),
+                    )
+                })?
+                .ok_or_else(|| {
+                    ActorError::new(
+                        ExitCode::ErrNotFound,
+                        format!("Invalid Verifier {:}", params.address),
+                    )
+                })?;
+
+            if verifier_cap >= Zero::zero() {
+                panic!("new verifier cap should be greater than or equal to 0");
+            }
+            let new_verifier_cap = verifier_cap - params.deal_size.clone();
+            if new_verifier_cap < MINIMUM_VERIFIED_SIZE.into() {
+                // Delete entry if remaining DataCap is less than MinVerifiedDealSize.
+                // Will be restored later if the deal did not get activated with a ProvenSector.
+                st.delete_verified_client(rt.store(), &params.address)
+                    .map_err(|_| {
+                        ActorError::new(
+                            ExitCode::ErrIllegalState,
+                            format!(
+                                "Failed to delete verified client{:} with bytes {:?}",
+                                params.clone().address,
+                                params.deal_size
+                            ),
+                        )
+                    })
+            } else {
+                st.put_verified_client(rt.store(), &params.address, &new_verifier_cap)
+                    .map_err(|_| {
+                        ActorError::new(
+                            ExitCode::ErrIllegalState,
+                            format!(
+                                "Failed to put verified client{:} with bytes {:}",
+                                params.address, params.deal_size
+                            ),
+                        )
+                    })
+            }
+        })??;
+
+        Ok(())
+    }
+
+    // Called by HandleInitTimeoutDeals from StorageMarketActor when a VerifiedDeal fails to init.
+    // Restore allowable cap for the client, creating new entry if the client has been deleted.
+    pub fn restore_bytes<BS, RT>(rt: &mut RT, params: UseBytesParams) -> Result<(), ActorError>
```

Comment:
```suggestion
    pub fn restore_bytes<BS, RT>(rt: &mut RT, params: RestoreBytesParams) -> Result<(), ActorError>
```

Looks like that type is already setup